kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -323,16 +326,20 @@ private AmazonS3 getConnection(String bucketName) {
                     .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(url, region)).build();
             // test connection once before returning it
             connection.doesBucketExistV2(bucketName);
-            retry = 0;
```

Comment:
If this is removed, will the retry of first time will not get added to this second time ?

The scenario is if maxRetry is 3, first time if the connection is received in second try, when the conneciton is received second time without cache, the retry left will be only 2.

Reply:
in line number 335 i am adding retry = 0, so when max retry is exceeded it will reset retry to 0 again and throw exception. it will reset connection to null as well.
Now in next call it will again try to get the connection and if failing again it will retry.